catkin_ws/src/vision/src/object_detection.py

```diff
@@ -79,85 +84,93 @@ def detect_on_image(raw_img, camera_id):
-                    obj_y.append(pred_obj_y)
-                    obj_z.append(pred_obj_z) 
-                    obj_theta_z.append(theta_z)
-                    extra_field.append(leftmost_gate_symbol) # 1 for earth, 0 for the other one
+                    detectionFrame.x = pred_obj_x
+                    detectionFrame.y = pred_obj_y
+                    detectionFrame.z = pred_obj_z
+                    detectionFrame.theta_z = theta_z
+                    detectionFrame.extra_field = leftmost_gate_symbol # 1 for earth, 0 for the other one
+
+                    detectionFrameArray.append(detectionFrame)
                 elif global_class_name == "Buoy": # BUOY
-                    label.append(global_class_name)
+                    detectionFrame.label = global_class_name
                     confidence.append(conf)
                     theta_z = measureAngle(bbox)
                     pred_obj_x, pred_obj_y, pred_obj_z = getObjectPositionFrontCam(bbox)
-                    obj_x.append(pred_obj_x)
-                    obj_y.append(pred_obj_y)
-                    obj_z.append(pred_obj_z) 
-                    obj_theta_z.append(theta_z)
-                    extra_field.append(None)
+                    detectionFrame.x = pred_obj_x
+                    detectionFrame.y = pred_obj_y
+                    detectionFrame.z = pred_obj_z
+                    detectionFrame.theta_z = theta_z
+                    detectionFrame.extra_field = None
+
+                    detectionFrameArray.append(detectionFrame)
 
                     for symbol_class_name, symbol_x, symbol_y, symbol_z, confidence in buoy_symbols:
-                        label.append(symbol_class_name)
-                        obj_x.append(symbol_x)
-                        obj_y.append(symbol_y)
-                        obj_z.append(symbol_z) 
+                        detectionFrame = VisionObject()
+                        detectionFrame.label = symbol_class_name
+                        detectionFrame.x = symbol_x
+                        detectionFrame.y = symbol_y
+                        detectionFrame.z = symbol_z
                         confidence.append(confidence)
-                        obj_theta_z.append(theta_z)
-                        extra_field.append(None)
-                elif global_class_name == "Octagon Table": # OCTAGON TABLE
-                    label.append(global_class_name)
-                    confidence.append(conf)
-                    pred_obj_x, pred_obj_y, pred_obj_z = getObjectPositionFrontCam(bbox)
-                    obj_x.append(pred_obj_x)
-                    obj_y.append(pred_obj_y)
-                    obj_z.append(pred_obj_z) 
-                    obj_theta_z.append(None)
-                    extra_field.append(None)
+                        detectionFrame.theta_z = theta_z
+                        detectionFrame.extra_field = None
+
+                        detectionFrameArray.append(detectionFrame)
```

Comment:
also this needs to be appended after the detectionFrame for the buoy, otherwise the order of the confidences will be messed up. Might make it easier if the confidence is an attribute of the VisionObject message and we can make sure they're always synced up. (sorry about that I only realized this looking at it now)

Reply:
I don't think it will mess anything because whenever I add a detectionFrame to the array I add the confidence too. So they are always in sync. 
But yeah adding the confidence as an attribute may be better. (It can be a good task for a new member)

Reply:
sorry about that you're right, good job👍